File: 1.foundation/1.hello-llm/javascript/Hellollm.js
// Hellollm.js

/**
 * Calls the Groq LLM API and logs the response.
 * Requires GROQ_API_KEY to be set in your environment variables.
 */
async function callGroqChat() {
  const apiKey = process.env.GROQ_API_KEY;
  if (!apiKey) {
    console.error("Error: GROQ_API_KEY environment variable not set.");
    return;
  }

  try {
  const response = await fetch("https://api.groq.com/openai/v1/chat/completions", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "Authorization": `Bearer ${apiKey}`
    },
    body: JSON.stringify({
      model: "llama-3.3-70b-versatile",
      messages: [
        { role: "user", content: "Hello LLM!" }
      ]
    })
  });

    if (!response.ok) {
      throw new Error(`API error: ${response.status} ${response.statusText}`);
    }

  const data = await response.json();
  console.log("Groq says:", data.choices[0].message.content);
  } catch (error) {
    console.error("Failed to call Groq API:", error.message);
  }
}

module.exports = { callGroqChat };

callGroqChat()

File: 1.foundation/1.hello-llm/java/HelloLlm.java
public class HelloLlm {
    public static void main(String[] args) throws Exception {
        String apiKey = System.getenv("GROQ_API_KEY");

        if (apiKey == null || apiKey.isEmpty()) {
            throw new IllegalArgumentException("GROQ_API_KEY environment variable is not set.");
        }

        String url = "https://api.groq.com/openai/v1/chat/completions";

        String requestBody = """
        {
            "model": "llama-3.3-70b-versatile",
            "messages": [
                { "role": "user", "content": "Hello LLM!" }
            ]
        }
        """;

        java.net.http.HttpClient client = java.net.http.HttpClient.newHttpClient();
        java.net.http.HttpRequest request = java.net.http.HttpRequest.newBuilder()
                .uri(java.net.URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .POST(java.net.http.HttpRequest.BodyPublishers.ofString(requestBody))
                .build();

        java.net.http.HttpResponse<String> response = client.send(request, java.net.http.HttpResponse.BodyHandlers.ofString());
        System.out.println("Response: " + response.body());
    }
}
